be/src/olap/types.h

```diff
@@ -255,6 +255,32 @@ struct BaseFieldtypeTraits : public CppTypeTraits<field_type> {
     }
 };
 
+template <typename T>
+OLAPStatus convert_int_from_varchar(void* dest, const void* src) {
+    using SrcType = typename CppTypeTraits<OLAP_FIELD_TYPE_VARCHAR>::CppType;
+    auto src_value = *reinterpret_cast<const SrcType*>(src);
```

Comment:
seems that this copy is not necessary. you can use input pointer to get_data or get_size